lib/vrl/parser/src/lex.rs

```diff
@@ -1903,4 +1919,26 @@ mod test {
             ],
         );
     }
+
+    #[test]
+    fn comment_in_block() {
+        test(
+            data(indoc!(
+                r#"if x {
+                      # It's an apostrophe.
+                      3
+                   }"#
+            )),
```

Comment:
When reading the other test cases in this file, I noticed that they have been intentionally kept on a single line, such that the `~` in the assertion line up neatly, e.g. 

https://github.com/vectordotdev/vector/blob/b9c0b9581b0b4db6022192c0cadf721426b2570d/lib/vrl/parser/src/lex.rs#L1771-L1789

Not perfect, but I'd suggest doing the same here:

```suggestion
            data(r#"if x {\n# It's an apostrophe.\n3\n}"#)),
```

Reply:
Have made the change, but it kinda goes wrong anyway since the `\n` pushes everything out by one char.

Reply:
Aaaah, I see... that's a shame. I wonder if we can fix that with reasonable effort. Would leave it like that for now, though.